src/core_ocean/shared/mpas_ocn_surface_bulk_forcing.F

```diff
@@ -198,11 +212,36 @@ subroutine ocn_surface_bulk_forcing_vel(meshPool, forcingPool, surfaceStress, su
       ! Convert coupled climate model wind stress to MPAS-O wind stress
       !$omp do schedule(runtime) private(cell1, cell2, zonalAverage, meridionalAverage)
       do iEdge = 1, nEdges
+        vertex1 = verticesOnEdge(1,iEdge)
+        vertex2 = verticesOnEdge(2,iEdge)
+
+        x2 = xVertex(vertex1)
+        y2 = yVertex(vertex1)
+        z2 = zVertex(vertex1)
+
+        x3 = xVertex(vertex2)
+        y3 = yVertex(vertex2)
+        z3 = zVertex(vertex2)
+
+        do iCell = 1,2
+          cell = cellsOnEdge(iCell, iEdge)
+          
+          x1 = xCell(cell)
+          y1 = yCell(cell)
+          z1 = zCell(cell)
+
+          areaWeights(iCell) = 0.5_RKIND*(sqrt(((y2-y1)*(z3-z1)-(y3-y1)*(z2-z1))**2+ &
```

Comment:
@sbrus89, the area of the "kite" between edges is `dcEdge*dvEdge`.  I think you could use this (or half of this) as your area and save a *lot* of computation.

Reply:
Thanks @xylar, good call.

Reply:
But, dcEdge is between cell centers.  So, this implicitly is averaging for the area computation, correct?

Reply:
I don't think we can simplify like this via `dcEdge*dvEdge`; we need the full triangle contained in the hexagon to be computed.

Reply:
Thanks @xylar!